### PR TITLE
Fix AC.GetAllConfigs

### PR DESCRIPTION
**Pull request by djmitche**
### What does this PR do?

This PR contains a fix to the GetAllConfigs method, which among other things is used by `agent check` to get autoconf config for the requested check.

This function had previously relied on a bug in AutoConfig whereby every call to `getAllConfigs` rescheduled all discovered configs, even if they had already been scheduled.  That bug was fixed with the reconciling config manager, causing GetAllConfigs to only return each config once.

### Motivation

`agent check xxx` doesn't work with the `logs_config.cca_in_ad` feature flag set.

### Additional Notes

Please review commit-by-commit.  In particular, the last commit removes a config parameter which I believe was never used, but I would appreciate additional eyes on that question.

### Describe how to test/QA your changes

*With `logs_config.cca_in_ad` both true and false*

Enable `logs_config.cca_in_ad` and run `agent check uptime`.  You should see check output.

Create a temporary config `conf.d/cpu.d/conf.yaml`:
```
ad_identifiers:
 - bash
init_config:
instances:
  - "tags": ["TEST:ING"]
```

Start a bash container (`docker run -ti --rm bash`) and run `agent check cpu`.  You should see check output.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by Kaderinho — COMMENTED** [review]
QA steps are working and I can correctly see the `TEST:INT` tag on the App. Thanks !

Some small notes : 

- With `cca_in_ad: true` : 
When you have some config issues, the following errors can be printed multiple times (2 in my case) when running `agent check ...`
```
error: Unable to load a check from instance of config '<name>': ...
error: Unable to load the check: unable to load any check from config '<name>'
```

- With `cca_in_ad: false` : 
Following errors appear for all checks that have already been scheduled :  
```
error: Unable to run Check <name>: a check with ID <name> is already running
```

Feel free to ignore if these output lines are acceptable. I'll let someone else approve it as I'm not really familiar with this part of code.

**Comment by djmitche** [comment]
Hm, it looks like maybe the checks are getting run "automatically" due to these changes.  I think we'll need some help from container-integrations to figure out how this should work.  It seems like it was a bit of a balancing act before?

File: pkg/autodiscovery/autoconfig.go [hunk 1]
```diff
@@ -208,12 +208,25 @@ func (ac *AutoConfig) ForceRanOnceFlag() {
...
+// elapsed.
+func (ac *AutoConfig) WaitUntilRunOnce() {
+	timeout := time.Millisecond * time.Duration(config.Datadog.GetInt("ac_load_timeout"))
+	now := time.Now()
+	for {
+		time.Sleep(100 * time.Millisecond) // don't hog the CPU
+		if ac.ranOnce.Load() {
+			return
+		}
+		if time.Since(now) > timeout {
```
Comment (clamoriniere) [inline pkg/autodiscovery/autoconfig.go]: could it be better to use a [context.WithTimeout](https://pkg.go.dev/context#WithTimeout) here?

File: pkg/autodiscovery/autoconfig.go [hunk 2]
```diff
@@ -198,34 +201,35 @@ func (ac *AutoConfig) AddConfigProvider(provider providers.ConfigProvider, shoul
 func (ac *AutoConfig) LoadAndRun() {
 	scheduleAll := ac.getAllConfigs()
 	ac.applyChanges(scheduleAll)
-	ac.ranOnce.Store(true)
+
 	log.Debug("LoadAndRun done.")
+
+	ac.m.Lock()
```
Comment (clamoriniere) [inline pkg/autodiscovery/autoconfig.go:207]: should the lock be at the start of the function? do we want to protect running several time `ac.applyChanges(scheduleAll)`?

**Comment by djmitche** [comment]
I put some notes in the Jira card, but I'm going to take a different approach to all of this.

The refactors of waiting for "run once" are probably not harmful, but ideally we'll eliminate this notion entirely, so there's no point in the refactors.

**Pull request closed without merging**